File: src/main/java/com/telegram/codex/telegram/CompactResultSender.java
package com.telegram.codex.telegram;

import com.telegram.codex.conversation.session.SessionCompactResult;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class CompactResultSender {

    private final TelegramClient telegramClient;

    public CompactResultSender(TelegramClient telegramClient) {
        this.telegramClient = telegramClient;
    }

    public void send(String chatId, SessionCompactResult result) {
        String text = switch (result.status()) {
            case MISSING_SESSION -> "而家冇 active session，冇嘢可以 compact。";
            case TOO_SHORT -> "目前對話得 " + result.messageCount() + " 段訊息，未去到要壓縮 context。";
            case OK -> String.join("\n",
                "已經將目前 session compact 成新 context。",
                "原本訊息：" + result.originalMessageCount(),
                "",
                result.compactText()
            );
        };
        telegramClient.sendMessage(chatId, text, List.of(), true);
    }
}

File: src/telegram/commands/compact_result_sender.rs
use std::sync::Arc;

use anyhow::Result;

use crate::conversation::session::session_service::{SessionCompactResult, SessionCompactStatus};
use crate::telegram::shared::telegram_types::TelegramGateway;

pub struct CompactResultSender {
    telegram_client: Arc<dyn TelegramGateway>,
}

impl CompactResultSender {
    pub fn new(telegram_client: Arc<dyn TelegramGateway>) -> Self {
        Self { telegram_client }
    }

    pub async fn send(&self, chat_id: &str, result: &SessionCompactResult) -> Result<()> {
        let text = match result.status {
            SessionCompactStatus::MissingSession => "而家冇 active session，冇嘢可以 compact。".to_owned(),
            SessionCompactStatus::TooShort => format!("目前對話得 {} 段訊息，未去到要壓縮 context。", result.message_count.unwrap_or(0)),
            SessionCompactStatus::Ok => [
                "已經將目前 session compact 成新 context。".to_owned(),
                format!("原本訊息：{}", result.original_message_count.unwrap_or(0)),
                String::new(),
                result.compact_text.clone().unwrap_or_default(),
            ]
            .join("\n"),
        };
        self.telegram_client.send_message(chat_id, Some(&text), &[], true).await
    }
}
